components/public-api-server/pkg/apiv1/tokens.go

```diff
@@ -198,23 +201,76 @@ func (s *TokensService) RegeneratePersonalAccessToken(ctx context.Context, req *
 }
 
 func (s *TokensService) UpdatePersonalAccessToken(ctx context.Context, req *connect.Request[v1.UpdatePersonalAccessTokenRequest]) (*connect.Response[v1.UpdatePersonalAccessTokenResponse], error) {
-	tokenID, err := validateTokenID(req.Msg.GetToken().GetId())
+	const (
+		nameField   = "name"
+		scopesField = "scopes"
+	)
+	var (
+		updatableMask = fieldmaskpb.FieldMask{Paths: []string{nameField, scopesField}}
+	)
+
+	tokenReq := req.Msg.GetToken()
+
+	tokenID, err := validateTokenID(tokenReq.GetId())
+	if err != nil {
+		return nil, err
+	}
+
+	mask, err := validateFieldMask(req.Msg.GetUpdateMask(), tokenReq)
 	if err != nil {
 		return nil, err
 	}
 
+	// If no mask fields are specified, we treat the request as updating all updatable fields
+	if len(mask.GetPaths()) == 0 {
+		mask = &updatableMask
+	}
+
 	conn, err := getConnection(ctx, s.connectionPool)
 	if err != nil {
 		return nil, err
 	}
 
-	_, _, err = s.getUser(ctx, conn)
+	_, userID, err := s.getUser(ctx, conn)
 	if err != nil {
 		return nil, err
 	}
 
-	log.Infof("Handling UpdatePersonalAccessToken request for Token ID '%s'", tokenID.String())
-	return nil, connect.NewError(connect.CodeUnimplemented, errors.New("gitpod.experimental.v1.TokensService.UpdatePersonalAccessToken is not implemented"))
+	toUpdate := fieldmaskpb.Intersect(mask, &updatableMask)
+	updateOpts := db.UpdatePersonalAccessTokenOpts{
+		TokenID: tokenID,
+		UserID:  userID,
+	}
+
+	fmt.Println("paths", toUpdate.GetPaths())
```

Comment:
Another `Println` that we can get rid of.

Reply:
Ah, yes, thanks. I forgot to remove them after some debugging.

Reply:
Thanks. fixed now.